### PR TITLE
[mlir] Attribute add printStripped

### PR DESCRIPTION
**Pull request by jpienaar**
Enable printing without dialect wrapping.

This closely matches `AsmPrinter::printStrippedAttrOrType` implementation wise except templating component.

### CODE REVIEW

**Comment by joker-eph** [comment]
Can we test this?

**Comment by jpienaar** [comment]
The easiest one I can think of is a plain unit test. I'll add one. (I mean the baseline functionality here is used when the dialect is omitted inside an op with region of same dialect, but fine to add one too :)).

File: mlir/unittests/IR/AttributeTest.cpp [hunk 1]
```diff
@@ -483,4 +483,22 @@ TEST(CopyCountAttr, CopyCount) {
...
 
+// Test stripped printing using test dialect attribute.
+TEST(CopyCountAttr, PrintStripped) {
+  MLIRContext context;
+  context.loadDialect<test::TestDialect>();
+  // Doesn't matter which dialect attribute is used, just chose TestCopyCount
+  // given proximity.
+  test::CopyCount::counter = 0;
+  test::CopyCount copyCount("hello");
+  auto res = test::TestCopyCountAttr::get(&context, std::move(copyCount));
```
Comment (joker-eph) [inline mlir/unittests/IR/AttributeTest.cpp:494]:
```suggestion
  Attribute res = test::TestCopyCountAttr::get(&context, std::move(copyCount));
```

We need to type-erased here, to differentiate with a concrete attribute.

**Review by joker-eph — APPROVED** [review]
LG, with a nit in the test.

File: mlir/unittests/IR/AttributeTest.cpp (hunk 1 above)
Comment (jpienaar) [reply mlir/unittests/IR/AttributeTest.cpp]: Good point, done. (and I missed that I could have just hit the "commit suggestion" button here ...)

**Pull request merged**